qiime2/sdk/action.py

```diff
@@ -155,6 +175,8 @@ def callable_wrapper(*args, **kwargs):
             for name in self.signature.inputs:
                 artifact = artifacts[name] = user_input[name]
                 provenance.add_input(name, artifact)
+                if self._is_subprocess():
+                    artifact._orphan()
```

Comment:
This works together with the new `__setstate__` in `Archiver` in ways I haven't fully grokked. For some reason when a worker receives the input, the atexit cleanup happens, but when the worker creates output, the atexit cleanup doesn't happen. 

I would have expected I could remove them both, but that doesn't work either.